cinema/models.py

```diff
@@ -84,23 +84,43 @@ class Ticket(models.Model):
     row = models.IntegerField()
     seat = models.IntegerField()
 
-    def clean(self):
-        for ticket_attr_value, ticket_attr_name, cinema_hall_attr_name in [
-            (self.row, "row", "rows"),
-            (self.seat, "seat", "seats_in_row"),
-        ]:
-            count_attrs = getattr(
-                self.movie_session.cinema_hall, cinema_hall_attr_name
+    @staticmethod
+    def validate_seat(
+            movie_session,
+            ticket_attr_value,
+            ticket_attr_name,
+            cinema_hall_attr_name,
+            error_to_raise
+    ):
+        count_attrs = getattr(
+            movie_session.cinema_hall,
+            cinema_hall_attr_name
+        )
+        if not (1 <= ticket_attr_value <= count_attrs):
+            raise error_to_raise(
+                {
+                    ticket_attr_name: f"{ticket_attr_name} "
+                                      f"number must be in available range: "
+                                      f"(1, {cinema_hall_attr_name}): "
+                                      f"(1, {count_attrs})"
+                }
             )
-            if not (1 <= ticket_attr_value <= count_attrs):
-                raise ValidationError(
-                    {
-                        ticket_attr_name: f"{ticket_attr_name} "
-                        f"number must be in available range: "
-                        f"(1, {cinema_hall_attr_name}): "
-                        f"(1, {count_attrs})"
-                    }
-                )
+
+    def clean(self):
+        Ticket.validate_seat(
+            self.movie_session,
+            self.row,
+            "row",
+            "rows",
+            ValidationError
+        )
+        Ticket.validate_seat(
```

Comment:
Why did you change this?

Reply:
to provide validation for creating tickets on serializer level